include/alpaka/mem/buf/oacc/Copy.hpp

```diff
@@ -284,92 +270,89 @@ namespace alpaka
         template<typename TDim>
         struct CreateTaskMemcpy<TDim, DevOacc, DevCpu>
         {
-            template<typename TExtent, typename TViewSrc, typename TViewDst>
+            template<typename TExtent, typename TViewSrc, typename TViewDstFwd>
             ALPAKA_FN_HOST static auto createTaskMemcpy(
-                TViewDst& viewDst,
+                TViewDstFwd&& viewDst,
                 TViewSrc const& viewSrc,
                 TExtent const& extent)
             {
                 ALPAKA_DEBUG_FULL_LOG_SCOPE;
 
-                return alpaka::oacc::detail::
-                    makeTaskCopyOacc<alpaka::oacc::detail::TaskCopyOacc, TDim, TViewDst, TViewSrc, TExtent>(
-                        viewDst,
-                        viewSrc,
-                        extent,
-                        getDev(viewDst),
-                        acc_memcpy_to_device);
+                auto devDst = getDev(viewDst);
+                return alpaka::oacc::detail::makeTaskCopyOacc<alpaka::oacc::detail::TaskCopyOacc, TDim>(
+                    std::forward<TViewDstFwd>(viewDst),
+                    viewSrc,
+                    extent,
+                    std::move(devDst),
```

Comment:
Is constructing `devDst` first and then moving it a style choice? Constructing it in-place would do the same.

Although, the move is superfluous, because `makeTaskCopyOacc` takes `devDst` by const reference.

Reply:
I changed `makeTaskCopyOacc` and the `TaskCopyOacc` to accept the device by value and move it into the class member.